### PR TITLE
Use simctl to take screenshots by default

### PR DESCRIPTION
**Pull request by dpgraham**

### CODE REVIEW

File: test/functional/basic/screenshot-e2e-specs.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  });
+
+
+  beforeEach(async () => {
+  });
+  afterEach(async () => {
+  });
+
+  it('should take screenshot', async () => {
+    await driver.takeScreenshot().should.eventually.be.a.string;
```
Comment (imurchie) [inline test/functional/basic/screenshot-e2e-specs.js]: https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/basic/basic-e2e-specs.js#L86-L111

File: lib/commands/screenshot.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.getScreenshot = async function () {
+  let screenshot;
+
+  try {
+    screenshot = await getScreenshot(this.opts.udid);
+  } catch (e) {
+    return this.proxyCommand('/screenshot', '/GET');
```
Comment (imurchie) [inline lib/commands/screenshot.js]: Some sort of logging of the error would be good, for debugging's sake.

Comment (imurchie) [inline lib/commands/screenshot.js]: Also, `return await`, or, to make it of the same sort as the previous call, `screenshot = await this.proxyCommand('/screenshot', '/GET');`.

File: test/functional/basic/screenshot-e2e-specs.js (hunk 1 above)
Comment (dpgraham) [inline test/functional/basic/screenshot-e2e-specs.js]: Aha, I'll move the tests over to that file

**Comment by dpgraham** [comment]
Changes applied

* Added logger that indicates when legacy screenshot is being used
* Removed 'screenshot.js' and moved everything into 'screenshots.js'
* Removed old test and just use the one that is already there
* Doesn't let it fail anymore, just checks that xcode version is >= 8.2

File: lib/commands/screenshots.js [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
 
 commands.getScreenshot = async function () {
   let data;
-  await retryInterval(10, 1000, async () => {
-    data = await this.proxyCommand('/screenshot', 'GET');
-    if (!_.isString(data)) {
-      throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
-    }
-  });
+  if (this.xcodeVersion.major > 8 || (this.xcodeVersion.major === 8 && this.xcodeVersion.minor >= 1)) {
```
Comment (imurchie) [inline lib/commands/screenshots.js]: This could also just be `if (this.xcodeVersion.versionFloat >= 8.1) {`.

**Comment by dpgraham** [comment]
Changes applied

**Comment by dpgraham** [comment]
Changes published to 2.25.1

**Pull request merged**